From 5e05d4b28c0fcf38e6df3dddb73754c9a8ea760c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:25 -0400
Subject: [usb] ehci-platform: Return immediately from suspend if ehci_suspend fails

Message-id: <1410887463-20674-374-git-send-email-dzickus@redhat.com>
Patchwork-id: 95541
O-Subject: [RHEL7 PATCH 373/591] usb: ehci-platform: Return immediately from suspend if ehci_suspend fails
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit e155b5b8d2d42455d3a94c2460c287e97184ec61
Author: Vivek Gautam <gautam.vivek@samsung.com>
Date:   Thu Apr 10 15:58:02 2014 +0530

    usb: ehci-platform: Return immediately from suspend if ehci_suspend fails

    Patch 'b8efdaf USB: EHCI: add check for wakeup/suspend race'
    adds a check for possible race between suspend and wakeup interrupt,
    and thereby it returns -EBUSY as error code if there's a wakeup
    interrupt.
    So the platform host controller should not proceed further with
    its suspend callback, rather should return immediately to avoid
    powering down the essential things, like phy.

    Signed-off-by: Vivek Gautam <gautam.vivek@samsung.com>
    Acked-by: Alan Stern <stern@rowland.harvard.edu>
    Cc: Hauke Mehrtens <hauke@hauke-m.de>
    Cc: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ehci-platform.c b/drivers/usb/host/ehci-platform.c
index d1f5cea..d2a00f6 100644
--- a/drivers/usb/host/ehci-platform.c
+++ b/drivers/usb/host/ehci-platform.c
@@ -175,6 +175,8 @@ static int ehci_platform_suspend(struct device *dev)
  int ret;
 
  ret = ehci_suspend(hcd, do_wakeup);
+ if (ret)
+  return ret;
 
  if (pdata->power_suspend)
   pdata->power_suspend(pdev);
-- 
1.7.1